File: server/src/database/database-handler.ts
import {MatchRecord, PlayerRecord} from "./records";
import {Match} from "../match";

export interface DatabaseHandler {
    fetchPlayers(): Promise<PlayerRecord[]>;
    fetchMatches(): Promise<MatchRecord[]>;
    fetchPlayerFromId(player: number): Promise<PlayerRecord>;
    createPlayer(name: string): Promise<void>;
    recordMatch(match: Match): Promise<void>;
}

File: kotlin-server/src/main/kotlin/com/example/kotlinserver/database/DatabaseHandler.kt
package com.example.kotlinserver.database

import com.example.kotlinserver.models.MatchModel
import com.example.kotlinserver.models.PlayerModel
import java.time.LocalDate

interface DatabaseHandler {
    fun fetchPlayers(): List<PlayerModel>
    fun fetchMatches(from: LocalDate, to: LocalDate): List<MatchModel>
    fun fetchPlayerFromId(playerId: Int): PlayerModel
    fun createPlayer(name: String)
    fun recordMatch(match: MatchModel)
}